subtests/docker_cli/systemd/systemd.py

```diff
@@ -0,0 +1,80 @@
+"""
+Summary
+-------
+
+Operational Summary
+---------------------
+    1. Copy Unit file
+    2. Build image using predefined Dockerfile & p4321-server.py
+    3. Run image
+    4. Confirm p4321-server script is running
+    5. Stop container
+    6. cleanup
+
+"""
+
+import shutil
+import time
+from os.path import join
+from autotest.client import utils
+from autotest.client.shared.utils import is_port_free
+from dockertest.subtest import Subtest
+from dockertest.dockercmd import DockerCmd
+
+
+class systemd(Subtest):
+
+    def initialize(self):
+        super(systemd, self).initialize()
+        self.stuff['tcp_ip'] = self.config['tcp_ip']
+        self.stuff['tcp_port'] = self.config['tcp_port']
+        self.stuff['sysd_unit_file'] = self.config['sysd_unit_file']
+        self.stuff['sysd_unitf_dest'] = self.config['sysd_unitf_dest']
+        self.stuff['name'] = self.config['name']
+        self.stuff['self.tmpdir'] = self.tmpdir
+        sysd_unitf = join(self.bindir, 'p4321.service')
+        dockerfilesrc = join(self.bindir, 'Dockerfile')
+        script_src = join(self.bindir, 'p4321-server.py')
+        shutil.copyfile(sysd_unitf, self.stuff['sysd_unitf_dest'])
+        shutil.copy(dockerfilesrc, self.stuff['self.tmpdir'])
+        shutil.copy(script_src, self.stuff['self.tmpdir'])
```

Comment:
Future maintainers will appreciate it if you condense the above two lines into:

    for build_file in ['Dockerfile', 'p4321-server.py']:
        shutil.copy(join(self.bindir, build_file), self.tmpdir)